src/Init/Data/BitVec/Lemmas.lean

```diff
@@ -3217,6 +3217,37 @@ theorem getElem_replicate {n w : Nat} (x : BitVec w) (h : i < w * n) :
   simp only [← getLsbD_eq_getElem, getLsbD_replicate]
   by_cases h' : w = 0 <;> simp [h'] <;> omega
 
+theorem replicate_append {x : BitVec w} :
+    x ++ x.replicate n = (x.replicate n ++ x).cast (by rw [Nat.add_comm]) := by
```

Comment:
Please replace this with
```
theorem append_assoc {x₁ : BitVec w₁} {x₂ : BitVec w₂} {x₃ : BitVec w₃} :
    (x₁ ++ x₂) ++ x₃ = (x₁ ++ (x₂ ++ x₃)).cast (by omega) := by
  sorry

theorem replicate_append {x : BitVec w} :
    x ++ x.replicate n = (x.replicate n ++ x).cast (by rw [Nat.add_comm]) := by
  induction n with
  | zero => simp
  | succ n ih =>
    rw [replicate_succ]
    conv => lhs; rw [ih]
    sorry -- just cleaning up casts.
```
Yesterday's general comment applies again. :-)

Reply:
The proof will be by induction on w\1, then using `rw [← cons_msb_setWidth x₁]`, and it seems a lemma `cons_append` is missing.